### ISSUE TITLE
add github issue option

### ISSUE THREAD

**Issue by mpadge** [mpadge/allcontributor#10]
Add option to `add_contributors()` function to open a github issue with dummy text to ping all contributors that they've been acknowledged in the README. Once that issue exists, triggering the option can then add any new contributors via additional automated issue comments.

**Comment by mpadge** [comment]
The above commit adds the initial functionality to open a new issue using the [github client](https://cli.github.com/).

## TODO

- [ ] Add function to add more recent contributors to an existing issue
- [ ] Add function to check whether `gh` is locally installed, and error if not

**Comment by mpadge** [comment]
Re-opening coz still have to do the 2nd of those, and also need to actually check that functionality works properly

**Comment by maelle** [comment]
Also, imagine I open an issue with a question and you end up telling me to rather open a discussion thread in a forum to get help. My contribution is then not really one. Could you label the issue as out-of-scope for the tool?

And regarding the original intent of this very issue, maybe it could be stated in the issue template instead?

**Comment by mpadge** [comment]
Yes, it would be very easy to label the issue as "out-of-scope" or "non-contrib" or something, but then again if issues really are not considered relevant contributions then they can simply be deleted. That was my initial thought at least, and the reason why i opted to retain a simpler acknowledge-all-type approach. A question is still a form of interest in, and presumably use of, the software. I guess for me the real issue in thinking about that was how such a label could be applied in a way that wasn't overtly negative. Label filtering in this context must necessarily be a *filtering out*, which is an inherently *negative* activity, so labels would work best if they contained a negation, but that is then potentially insulting. I tried but couldn't think of suitable non-negative labels that retained general applicability ("enquiry" comes close, but even then issues may be opened simply to comment without enquiry).

Any advice would be greatly appreciated! The whole issue extraction is graphql, so adding a labels field to the query would be totally easy, and this would be no trouble at all to implement. I just would like to do it in the most appropriate way possible.

----

Finally, I'm not sure exactly what you mean by this:

> And regarding the original intent of this very issue, maybe it could be stated in the issue template instead?

Could you please explain a bit more? Thanks

**Comment by maelle** [comment]
how would you delete an issue?

I was thinking the issue template could state there's a README section with contributions but then who reads that...

**Comment by mpadge** [comment]
Magic new-ish "delete issue" button:

![image](https://user-images.githubusercontent.com/6697851/98244335-91084000-1f6f-11eb-8f34-7698a42b66be.png)

> I was thinking the issue template could state there's a README section with contributions ...

I think that is what it already does:
https://github.com/mpadge/allcontributors/blob/126be32f5b51136099c2b176fd73253b139c4bb4/R/add-contributors.R#L377-L382

That's what you mean, right?

**Comment by maelle** [comment]
aaah cool that one can delete issues. 

by issue template I mean .github/ templates, that one sees when opening an issue.
